### PR TITLE
[#960] feat(dashboard): Uniffle needs to add a WebUI page

### PR DESCRIPTION
**Pull request by yl09099**
### What changes were proposed in this pull request?

Uniffle does not have a webUI interface for viewing parameters and absorbing relationships, so it needs a more intuitive display

### Why are the changes needed?

Fix: #960

### Does this PR introduce _any_ user-facing change?

Added a webUI interface to change the API access to Jersey

### How was this patch tested?

Add Some UnitTest

### CODE REVIEW

**Comment by yl09099** [comment]
unifflecoor
uniffleshuffle
coorapp

**Comment by yl09099** [comment]
The following changes were made:
1. VUE3+element plus was adopted to realize front-end related development;
2, the backend introduced Jersey support RESTful interface, learning from PR: #864 related code, but #864 Jersey will throw an exception, which has been fixed here;
3, at present, the relevant functions have not been developed, but the framework is introduced and the relevant front and back end can call and display the page;
4. After the PR: #864 merger is completed, I will continue to make changes

**Comment by yl09099** [comment]
3. at present, the relevant functions have not been developed, but the framework is introduced and the relevant front and back end can call and display the page;
4. After the PR: #864 merger is completed, I will continue to make changes

**Comment by yl09099** [comment]
5. For the information to be displayed on the page, you are welcome to make suggestions

**Comment by jerqi** [comment]
Could you write a document like https://docs.google.com/document/d/1G0cOFVJbYLf2oX1fiadh7zi2M6DlEcjTQTh4kSkb0LA/edit#heading=h.kl0vgpq0nxcy?
You should also give others authority to edit the document.
If we have a document, it will be more convenient to discuss.

**Comment by yl09099** [comment]
> the ci is failed and I believe we should solve jersey conflict first. Will review once the conflict is resolve

OK, I'll do it this week

**Comment by zuston** [comment]
> Could you write a document like https://docs.google.com/document/d/1G0cOFVJbYLf2oX1fiadh7zi2M6DlEcjTQTh4kSkb0LA/edit#heading=h.kl0vgpq0nxcy? You should also give others authority to edit the document. If we have a document, it will be more convenient to discuss.

+1 

I think the following points should be paid attention to and attached into design doc.
1. The dashboard project code structure
2. The restful api communication process, including shuffle-server/coordinator/dashboard server and the deployment design
3. Detailed screenshot about main pages

**Comment by yl09099** [comment]
https://docs.google.com/document/d/10IEbbo_xQS-GHy0ezRcm9mT4Fv9hqghDsXivDpK1kas/edit?usp=sharing @jerqi @advancedxy @zuston

**Comment by jerqi** [comment]
> https://docs.google.com/document/d/10IEbbo_xQS-GHy0ezRcm9mT4Fv9hqghDsXivDpK1kas/edit?usp=sharing @jerqi @advancedxy @zuston

Could you give us the authority of edition?

**Comment by yl09099** [comment]
> 

Document editing permission has been opened

**Comment by jerqi** [comment]
> > 
> 
> Document editing permission has been opened

Some comments left.

**Comment by yl09099** [comment]
> @yl09099 Hi, why this PR is closed?

I need to wait for them to resolve the Jersey conflict, I will temporarily close it, a few days I will submit new code, I will open again

**Comment by yl09099** [comment]
@advancedxy @jerqi I have developed the first version, please help me review it.

File: dashboard/src/main/webapp/jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (jerqi) [inline dashboard/src/main/webapp/jsconfig.json:1]: Every file must have Apache license.

**Review by jerqi — COMMENTED** [review]
Should we merge this pr after we choose another new logo for our project?

File: dashboard/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+              <goal>install-node-and-npm</goal>
+            </goals>
+            <phase>generate-resources</phase>
+            <configuration>
+              <nodeVersion>v20.2.0</nodeVersion>
+              <npmVersion>9.6.6</npmVersion>
+              <nodeDownloadRoot>https://nodejs.org/dist/</nodeDownloadRoot>
+            </configuration>
+          </execution>
+          <!-- 执行脚本，删除node_modules和package-lock.json -->
```
Comment (jerqi) [inline dashboard/pom.xml:56]: Could we use English comments?

File: dashboard/pom.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            <id>npm install</id>
+            <goals>
+              <goal>npm</goal>
+            </goals>
+            <phase>generate-resources</phase>
+            <configuration>
+              <arguments>install</arguments>
+            </configuration>
+          </execution>
+          <!-- build 之后复制文件到 src/main/resource/static 下 -->
```
Comment (jerqi) [inline dashboard/pom.xml:78]: ditto.

**Comment by jerqi** [comment]
Should we the default configuration file to the directory `conf`? Should we add some documents to explain how to use dashboard. We can add the document in the next pr surely.

**Comment by yl09099** [comment]
> Should we the default configuration file to the directory `conf`? Should we add some documents to explain how to use dashboard. We can add the document in the next pr surely.

Ok, I'll correct it right away

**Comment by yl09099** [comment]
> 

The new icon is already being designed, I will modify it next time

**Comment by xianjingfeng** [comment]
Do we must deploy dashboard as a single process? Could we deploy it by exists jetty server that in coordinator？

**Comment by yl09099** [comment]
> Do we must deploy dashboard as a single process? Could we deploy it by exists jetty server that in coordinator？

Users can choose whether to start the dashboard module, which cannot be implemented without it

**Comment by xianjingfeng** [comment]
> > Do we must deploy dashboard as a single process? Could we deploy it by exists jetty server that in coordinator？
> 
> Users can choose whether to start the dashboard module, which cannot be implemented without it

Why do users not use dashboard? What are the possible reasons?
We need to deploy one more process,  and we need to set the configuration for it, and we also need to consider how to deploy it on kubernetes.
And in the future, we need to do permission check on both the coordinator and the dashboard.

**Comment by jerqi** [comment]
> > > Do we must deploy dashboard as a single process? Could we deploy it by exists jetty server that in coordinator？
> > 
> > 
> > Users can choose whether to start the dashboard module, which cannot be implemented without it
> 
> Why do users not use dashboard? What are the possible reasons? We need to deploy one more process, and we need to set the configuration for it, and we also need to consider how to deploy it on kubernetes. And in the future, we need to do permission check on both the coordinator and the dashboard.

If we do some slow analysis operations in the dashboard, it will influence the coordinator if dashboard is deployed in the same process as the coordinator. It's ok to deploy them in two processes.

It's ok if users want to deploy them in one process. You provide an option to support the feature.

**Comment by xianjingfeng** [comment]
> If we do some slow analysis operations in the dashboard, it will influence the coordinator if dashboard is deployed in the same process as the coordinator. It's ok to deploy them in two processes.
> 
> It's ok if users want to deploy them in one process. You can provide an option to support the feature.

Deploying in the same process can also achieve this feature. Just add a switch for the following code
https://github.com/apache/incubator-uniffle/blob/78a8115b989d68696e5d8f473b633f3cb034d692/dashboard/src/main/java/org/apache/uniffle/dashboard/web/JettyServerFront.java#L90-L92

**Pull request closed without merging**